clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp

```diff
@@ -2368,10 +2369,20 @@ mlir::Value CIRGenFunction::buildCommonNeonBuiltinExpr(
                        : "llvm.aarch64.neon.srhadd";
     break;
   }
+  case NEON::BI__builtin_neon_vqmovun_v: {
+    intrincsName = "llvm.aarch64.neon.sqxtun";
+    argTypes.push_back(builder.getExtendedOrTruncatedElementVectorType(
+        vTy, true /* extended */, true /* signed */));
+    break;
+  }
+  }
+
+  if (!intrincsName.empty()) {
```

Comment:
I'd revert this logic to make things a bit more direct:
```
if (intrincsName.empty())
  return nullptr;
// rest of code that assumes it exists.
```

Reply:
Will fix it. And after reading LLVM coding convention throughly on airplane recently, I can tell this suggestion is in the spirit of [ this part of llvm coding convention](https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code ). You certainly already have the convention ingrained in your mind  :-)